### PR TITLE
(MODULES-8193) - Removal of inbuilt deepmerge and dirname functions

### PR DESCRIPTION
**Pull request by david22swan**
Both functions have been removed from the code and their function calls within the manifest's updated as they already exist within puppet itself, in the case of deepmerge and within stdlib in the case of dirname.

### CODE REVIEW

File: manifests/server/config.pp [hunk 1]
```diff
@@ -43,9 +43,10 @@
 
     # on mariadb systems, $includedir is not defined, but /etc/my.cnf.d has
     # to be managed to place the server.cnf there
-    $configparentdir = mysql::dirname($mysql::server::config_file)
+    $configparentdir = dirname($mysql::server::config_file)
     if $configparentdir != '/' and $configparentdir != '/etc' and $configparentdir
-        != $includedir and $configparentdir != mysql::dirname($includedir) {
+        != $includedir and $includedir != undef and $includedir != '' and $configparentdir
+        != dirname($includedir) {
```
Comment (tphoney) [inline manifests/server/config.pp]: this is hideously hard to read. can it be simplified or split up

**Comment by david22swan** [comment]
Have separated/updated code in such as way as to improve readability while maintaining the original functionality of the code.

**Comment by david22swan** [comment]
@tphoney Are you fine with the code as it is now?

**Comment by david22swan** [comment]
screen shot 2018-12-13 at 12 04 26 pm

Changes run green on adhoc

**Pull request merged**